### DISCUSSION TITLE
Consolidation of external /  local ID abstraction into Universal HolonId

### DISCUSSION THREAD

**Discussion by nphias** [evomimic/map-holons#194] [Design Discussions]
in Essense I am talking about one universal ID and suggesting we do not make any distinction between local/external etc..
an ID can be universally the DNA hash of the space and the action hash of the holon entry.
we know that the DNA bundle represents a unique DHT confirmed by the hash

given that we are saying that a "Space is equal to a DHT"

keeping the struct allows us to break parts of the holonid, so ie:
```
pub struct HolonId {
    pub space_id: DNAHash
    pub address_id: ActionHash
}
```

**Comment by evomimic** [comment]
Please clarify what problem is being addressed. In what way is the definition of HolonId "convoluted"?

The only thing being passed in function arguments today is HolonId, so the proposed solution doesn't change anything as far as usage. 

Dropping the enum just makes it harder for implementation logic to distinguish local from external id's. But this distinction matters greatly. 

In short, the proposal would seem to have no impact on usage and a negative impact on implementation. What am I missing?
